src/cheri_pte.sail

```diff
@@ -38,102 +38,69 @@ union PTE_Check = {
+  let lcav = if e.LoadCap()  == 0b1 then PTW_LCM_OK else PTW_LCM_CLR;
+  let scav = if e.StoreCap() == 0b1 then PTW_SCM_OK else PTW_SCM_TRAP;
+  let (succ, ext_ptw') : (bool, ext_ptw) =
   match (ac, priv) {
-    (Read(Data),      User)       => to_pte_check(p.U() == 0b1 & (p.R() == 0b1 | (p.X() == 0b1 & mxr))),
-    (Write(Data),     User)       => to_pte_check(p.U() == 0b1 & p.W() == 0b1),
-    (ReadWrite(Data), User)       => to_pte_check(p.U() == 0b1 & p.W() == 0b1 & (p.R() == 0b1 | (p.X() == 0b1 & mxr))),
-    (Execute(),       User)       => to_pte_check(p.U() == 0b1 & p.X() == 0b1),
+    (Read(Data),      User)       => (p.U() == 0b1 & (p.R() == 0b1 | (p.X() == 0b1 & mxr)), ext_ptw),
+    (Write(Data),     User)       => (p.U() == 0b1 & p.W() == 0b1, ext_ptw),
+    (ReadWrite(Data), User)       => (p.U() == 0b1 & p.W() == 0b1 & (p.R() == 0b1 | (p.X() == 0b1 & mxr)), ext_ptw),
+    (Execute(),       User)       => (p.U() == 0b1 & p.X() == 0b1, ext_ptw),
 
-    (Read(Data),      Supervisor) => to_pte_check((p.U() == 0b0 | do_sum) & (p.R() == 0b1 | (p.X() == 0b1 & mxr))),
-    (Write(Data),     Supervisor) => to_pte_check((p.U() == 0b0 | do_sum) & p.W() == 0b1),
-    (ReadWrite(Data), Supervisor) => to_pte_check((p.U() == 0b0 | do_sum) & p.W() == 0b1 & (p.R() == 0b1 | (p.X() == 0b1 & mxr))),
-    (Execute(),       Supervisor) => to_pte_check(p.U() == 0b0 & p.X() == 0b1),
+    (Read(Data),      Supervisor) => ((p.U() == 0b0 | do_sum) & (p.R() == 0b1 | (p.X() == 0b1 & mxr)), ext_ptw),
+    (Write(Data),     Supervisor) => ((p.U() == 0b0 | do_sum) & p.W() == 0b1, ext_ptw),
+    (ReadWrite(Data), Supervisor) => ((p.U() == 0b0 | do_sum) & p.W() == 0b1 & (p.R() == 0b1 | (p.X() == 0b1 & mxr)), ext_ptw),
+    (Execute(),       Supervisor) => (p.U() == 0b0 & p.X() == 0b1, ext_ptw),
 
     (Read(Cap),       User)       => { /* load-cap should probably not be affected by mxr. */
                                        if   p.U() == 0b1 & p.R() == 0b1
-                                       then {
-                                         let  e = Mk_Ext_PTE_Bits(ext);
-                                         if   e.LoadCap() == 0b1
-                                         then (true, PTE_CAP_OK)
-                                         else {
-                                           /* Allow the address translation to proceed, but mark for tag stripping */
-                                           (true, PTE_LOAD_CAP_ERR)
-                                         }
-                                       }
-                                       else (false, PTE_CAP_OK)
+                                       then
+                                         /*
+                                          * Allow the address translation to proceed,
+                                          * but mark for tag stripping
+                                          */
+                                         (true, ext_ptw_join_lcm(ext_ptw, lcav))
```

Comment:
```suggestion
                                        /*
                                         * Allow the address translation to proceed,
                                         * but mark for tag stripping
                                         */
                                        then (true, ext_ptw_join_lcm(ext_ptw, lcav))
```
and other instances

Reply:
I'd dropped these comments when I did some of my own tidying, it seems, so they're no longer present in the version just pushed.